base/src/main/java/org/kontinuity/catapult/base/identity/IdentityVisitor.java

```diff
@@ -0,0 +1,15 @@
+package org.kontinuity.catapult.base.identity;
+
+/**
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+public interface IdentityVisitor {
+
+    default void visit(TokenIdentity token) {
```

Comment:
Within an IdentityVisitor implementation, you only implement what you need, otherwise an UOE is thrown. That´s why I declared `default` methods. I interpret UOE as something wrong in code (calling an unsupported API), which is exactly what may happen (eg. Authentication with Token where only user/pass is supported should throw a UOE)